xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/xwiki.js

```diff
@@ -1992,3 +1992,43 @@ document.observe("xwiki:dom:loaded", function() {
   interceptXMLHttpRequest();
   interceptFetch();
 })();
+
+/**
+ * Initializes the submit-once protection on the current page.
+ * Search for all form elements with a .xwiki-submit-once class one the document (or on an updated sub-part of the dom).
+ * For each form, once the form is submitted, mark it with the .xwiki-form-submitting class and prevent new submissions.
+ *
+ *  @since 14.10.17
+ *  @since 15.5.3
+ *  @since 15.8-rc-1
+ */
+document.observe('xwiki:dom:loaded', function () {
```

Comment:
Shouldn't we use modern JavaScript APIs or at least jQuery instead of Prototype.js?